src/index.js

```diff
@@ -611,6 +611,10 @@ class Analytics {
       return Promise.resolve();
     }
 
+    // remove items from the queue before awaiting for the pending flush
+    // to avoid concurrent events triggering unnecessary flush API calls
+    const items = this.queue.splice(0, this.flushAt);
+
```

Comment:
We should avoid this. If the pending flush throws an error, we'll lose these events from the queue.
Instead, restore Ln 623 and check for the length of `items` before proceeding further. Make sure to invoke the callback when it's empty.

Reply:
Adressed